inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp

```diff
@@ -7,17 +7,84 @@
 #include "functional_test_utils/layer_test_utils.hpp"
 #include "low_precision_transformations/transformer.hpp"
 
+#include "ie_util_internal.hpp"
+#include "low_precision_transformations/convolution.hpp"
+#include "low_precision_transformations/scaleshift_to_convolution.hpp"
+
 namespace LayerTestsUtils {
 
-class LayerTransformation : public testing::WithParamInterface<LayerTestsUtils::basicParams>, public LayerTestsUtils::LayerTestsCommon {
+typedef std::tuple<
+    InferenceEngine::Precision,
+    InferenceEngine::SizeVector,
+    std::string,
+    InferenceEngine::details::LayerTransformation::Params> LayerTransformationParams;
+
+class LayerTransformationParamsFactory {
+public:
+    static InferenceEngine::details::LayerTransformation::Params createParamU8I8();
+    static InferenceEngine::details::LayerTransformation::Params createParamU8U8();
+    static InferenceEngine::details::LayerTransformation::Params createParamI8I8();
+    static InferenceEngine::details::LayerTransformation::Params createParamCpu();
+    static InferenceEngine::details::LayerTransformation::Params createParamGpu();
+};
+
+template <typename T>
+class LayerTransformation : public testing::WithParamInterface<T>, public LayerTestsUtils::LayerTestsCommon {
 public:
     InferenceEngine::details::LowPrecisionTransformations getLowPrecisionTransformations(
-        const InferenceEngine::details::LayerTransformation::Params& params) const;
+        const InferenceEngine::details::LayerTransformation::Params& params) const {
+        if (targetDevice == "CPU") {
```

Comment:
Why we use device separation in common utils lib?

Reply:
good comment, thanks: 32278: [LPT] [TEST] Move hardware specific logic to parent classes